### PR TITLE
Fixed: [UI] Add support for renaming glossary #8733

### LINKED ISSUE #8733: [UI] Add support for renaming glossary
**Feature**
#8438

**Describe the task**
Add support for renaming glossary and glossary terms. See related issue #7524

@harshach for release 0.13.1

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

Closes #8733 
- Added support to change name along with displayname

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx [hunk 1]
```diff
@@ -250,6 +250,7 @@ const GlossaryV1 = ({
     updatedDetails = {
       ...selectedData,
       displayName: displayName?.trim(),
+      name: displayName?.trim() || selectedData.name,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:253]: Is this intentional? passing displayName for the name.

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 2]
```diff
@@ -482,6 +482,22 @@ const GlossaryPageV1 = () => {
...
+      setGlossariesList([]);
+      setIsLoading(true);
+      history.push(getGlossaryPath());
+      fetchGlossaryList();
+    } else {
+      history.push(getGlossaryPath(redirectFqn));
+      fetchGlossaryList(redirectFqn);
+    }
+  };
+
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:500]: Can we add some comments for this method?

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx [hunk 3]
```diff
@@ -514,6 +530,9 @@ const GlossaryPageV1 = () => {
             }
           });
         });
+        if (selectedData?.name !== updatedData.name) {
+          handleRedirection(response.fullyQualifiedName);
+        }
       } else {
         throw jsonData['api-error-messages']['update-description-error'];
       }
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:538]: Is this the correct error message for the operation?

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:500]: Hi @Sachin-chaurasiya method name is self-explanatory, its just redirects to glossary

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 3 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:538]: hmm nice catch, seems like it is not!

File: openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx (hunk 2 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx:500]: @ShaileshParmar11 , Can we add comments for this logic?
```
const redirectFqn = fqn
      ? fqn
      : selectedKey.split('.').slice(0, -1).join('.');
```

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:253]: yes!

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:253]: That's the main goal of PR :)

**Comment by ShaileshParmar11** [comment]
Hello @Sachin-chaurasiya I have updated code!

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**